homeassistant/components/switchbee/button.py

```diff
@@ -0,0 +1,50 @@
+"""Support for SwitchBee scenario button."""
+
+from switchbee.api import SwitchBeeError
+from switchbee.device import ApiStateCommand, DeviceType, SwitchBeeBaseDevice
+
+from homeassistant.components.button import ButtonEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import SwitchBeeCoordinator
+from .const import DOMAIN
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up Switchbee button."""
+    coordinator: SwitchBeeCoordinator = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities(
+        SwitchBeeButton(switchbee_device, coordinator)
+        for switchbee_device in coordinator.data.values()
+        if switchbee_device.type == DeviceType.Scenario
+    )
+
+
+class SwitchBeeButton(CoordinatorEntity[SwitchBeeCoordinator], ButtonEntity):
+    """Representation of an Switchbee button."""
+
+    def __init__(
+        self,
+        device: SwitchBeeBaseDevice,
+        coordinator: SwitchBeeCoordinator,
+    ) -> None:
+        """Initialize the Switchbee switch."""
+        super().__init__(coordinator)
+        self._attr_name = f"{device.zone} {device.name}"
+        self._device_id: int = device.id
```

Comment:
The type hint is not needed here, as it is (or will be) inferred from the library.
```suggestion
        self._device_id = device.id
```